### DISCUSSION TITLE
Does `hasura migrate apply` require the hasura server to be running or just the database?

### DISCUSSION THREAD

**Discussion by samuela** [hasura/graphql-engine#6451] [Question] [answered]
Do `hasura migrate apply` and `hasura metadata apply` require the hasura server to be running or just the database? I'm trying to implement a continuous delivery pipeline such that my app will be re-deployed on every commit to the main branch. In this case, I'm trying to deploy hasura to render.com, and it's proven to be a bit messy. The [docs](https://hasura.io/docs/1.0/graphql/core/deployment/deployment-guides/render-one-click.html) only illustrate how to deploy hasura in "mutable" mode.

How have you guys built CD workflows with hasura?  Does anyone have any tips?

**Comment by tirumaraiselvan** [comment, marked as answer]
Yes, it requires the server to be running. Hasura provides a pure SQL API that runs the migrations and then applies the metadata via another API.
